### ISSUE TITLE
[ENH] wishlist for de-novo implementations of torch based models

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#6660]
This issue collects a wishlist of de-novo implementations of torch based models.
Anyone can suggest models to implement, we will have to prioritize along a impact/cost analysis.

FYI @fnhirwa.

**Comment by fkiraly** [comment]
### Forecasting

suggested by @corradomio in https://github.com/sktime/sktime/issues/6656

- nhits
- beats
- transformer family - autoformer etc https://github.com/sktime/sktime/issues/4939

###Time series classification and regerssion

suggestions from @fnhirwa's post https://github.com/sktime/sktime/issues/6641

- [ ] GRU (Cho, 2014) [Paper](https://arxiv.org/abs/1412.3555)
- [ ] GRU-FCN(Elsayed, 2018)[Paper](https://arxiv.org/abs/1812.07683)
- [ ] TabTransformer (Huang, 2020) [Paper](https://arxiv.org/pdf/2012.06678)
- [ ] LSTM-FCN (Karim, 2017) [Paper](https://arxiv.org/abs/1709.05206)
- [ ] MLSTM-FCN - Multivariate LSTM-FCN (Karim, 2019) [Paper](https://arxiv.org/abs/1801.04503)
- [ ] TST - Time Series Transformer (Zerveas, 2020) [Paper](https://dl.acm.org/doi/abs/10.1145/3447548.3467401)

**Comment by fkiraly** [comment]
@fnhirwa, could you reply so I can assign you?
Assignment is for consolidating (not completing) the wishlist.

**Comment by benHeid** [comment]
I think nbeats and nhits are available by PyTorch forecasting and autoformer is available via hugginface.

Additional models are:
* ES-RNN -> Winning method of M4 Challenge (I think Slawek Smyl published a new version)
* State space models, could be nice for time series forecasting (not sure if there exists forecasting methods, but should be easily applicable to forecasting)
* xLSTM based forecasting method? -> Probably there is no existing one, but could be cool to implement one.

**Comment by fnhirwa** [comment]
> @fnhirwa, could you reply so I can assign you? Assignment is for consolidating (not completing) the wishlist.

yeah you can assign mw to the issue

**Comment by corradomio** [comment]
There is a consideration to do:
1) sktime uses pandas Series & DataFrame
2) Seres & DataFrame are wrapper on numpy arrays
3) Pytorch (for noe we can exclue Tensorflow, BUT the reasoning doesnt't cnage) uses Tensors, an object a little more complex than an array nupy because it must save the gradients generated in the computation.
4) these tensors can be located in memory OR inside the GPU memory

Now, all 3rd NN libraries for TS (GluonTS, NeuralForecast, Darts) create ANOTHER wrapper above the Series/DataFrame.

To use a NN library, to "normalize" the code organization, it is used a framework as Pytorch Lightning or skorch.
The usefulness of these libraries is they ability to move the data inside the GPU using just a single flag.

The previous libraries use one of these frameworks (NeuralForecast uses Lightning) or a custom framework.

5) to "wrap" these libraries to be usable in sktime introduces ANOTHER layer.

I think it is better to reduce the number of layers.

A) sktime
B) a simple wrapper with the responsibility to handle the numpy arrays vs the NN tensors (for example skorch)
c) the NN model

**Comment by fkiraly** [comment]
hm, that's a very good point, @corradomio. `sktime` delegates the data wrapping to the boilerplate layer, and it can be turned entirely off if the user wants to.

@fnhirwa, is this a good idea for the refactor issue: https://github.com/sktime/sktime/issues/6641, on how we handle pytorch datasets, as well as tensors? FYI also @geetu040.

**Comment by fkiraly** [comment]
> create ANOTHER wrapper above the Series/DataFrame.

You are right, at some point it will catch fire if we are not careful😄

![ic55rwqnrip51](https://github.com/sktime/sktime/assets/7985502/d517617f-e03e-4e3d-9edb-830910a16de8)

(from the scarytechnology reddit https://www.reddit.com/r/ScaryTechnology/comments/j0953p/adapters_chain/)

**Comment by fkiraly** [comment]
Suggestion by Valeriy:
Transformer Conformal Prediction for Time Series: https://github.com/Jayaos/TCPTS

**Comment by fnhirwa** [comment]
> hm, that's a very good point, @corradomio. `sktime` delegates the data wrapping to the boilerplate layer, and it can be turned entirely off if the user wants to.
> 
> @fnhirwa, is this a good idea for the refactor issue: #6641, on how we handle pytorch datasets, as well as tensors? FYI also @geetu040.

This is a very sensible point; if we keep adding more abstraction layers, we might have unnecessary complexity at some point. I agree that this should be taken into consideration.

**Comment by fkiraly** [comment]
deassigning @fnhirwa since the task was "collecting", which seems reasonably complete. Assigning "good first issue" now, for people to implement or integrate.

**Comment by geetu040** [comment]
> xLSTM based forecasting method? -> Probably there is no existing one, but could be cool to implement one.

@fkiraly can you link this issue here with `xLSTM`
https://github.com/sktime/sktime/issues/6793 . [code](https://github.com/muslehal/xLSTMTime) . [paper](https://arxiv.org/pdf/2407.10240)

**Comment by fkiraly** [comment]
done

**Comment by geetu040** [comment]
Another LSTM-based model that can be added

LSTM-autoencoder with attentions for multivariate time series . [code](https://github.com/JulesBelveze/time-series-autoencoder) . [paper](https://arxiv.org/abs/1704.02971)

**Comment by fnhirwa** [comment]
I am starting to implement GRU (Cho, 2014) and GRU-FCN(Elsayed, 2018)

**Comment by sanskarmodi8** [comment]
Hey @fkiraly, can I work on implementing TST Classifier?

**Comment by sanskarmodi8** [comment]
Hey @fkiraly,
Can I work on implementing LSTM-autoencoder with attentions for multivariate time series?

**Comment by fkiraly** [comment]
sure!
